tests/std/tests/GH_001914_Cached_position/test.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <ranges>
+
+#include <range_algorithm_support.hpp>
+using namespace std;
+
+template <class View>
+class test_cache : public ranges::_Cached_position<View, test_cache<View>> {
+public:
+    constexpr test_cache(View view) : _view(move(view)) {}
+
+    constexpr void set_cache() {
+        this->_Set_cache(_view, _view.begin());
+    }
+
+    _NODISCARD constexpr bool has_cache() const noexcept {

```

Comment:
Should just be `[[nodiscard]]` in test code (the macro is only for product code).